examples/reminders/lib/base/data_sources/remote/reminders_firebase_data_source.dart

```diff
@@ -0,0 +1,163 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:rx_bloc_list/models.dart';
+
+import '../../models/reminder/reminder_list_reponse.dart';
+import '../../models/reminder/reminder_model.dart';
+import 'reminders_data_source.dart';
+
+class RemindersFirebaseDataSource implements RemindersDataSource {
+  RemindersFirebaseDataSource() {
+    remindersReference = fireStore.collection('reminders');
+  }
+
+  final fireStore = FirebaseFirestore.instance;
+  late CollectionReference remindersReference;
+
+  ///todo remove it from here
+  int remindersCollectionLength = 0;
+  QueryDocumentSnapshot? lastFetchedRecord;
+
+  late final List<ReminderModel> _data;
+
+  @override
+  Future<ReminderModel> create(
+      {required String title,
+      required DateTime dueDate,
+      required bool complete}) async {
+    var reminder = ReminderModel(
+      dueDate: dueDate,
+      id: remindersReference.doc().id,
+      title: title,
+      complete: complete,
+    );
+
+    var createdReminder = await remindersReference.add(reminder.toJson());
+    var createdReminderId = createdReminder.id;
+    reminder = reminder.copyWith(id: createdReminderId);
+    await remindersReference.doc(createdReminderId).update(reminder.toJson());
+
+    return reminder;
+  }
+
+  @override
+  Future<void> delete(String id) async {
+    await remindersReference.doc(id).delete();
+  }
+
+  @override
+  Future<ReminderListResponse> getAll(ReminderModelRequest? request) async {
+    //  Generate a query
+    var querySnapshot = getFirebaseFilteredQuery(request);
+
+    // Modify the query
+    if (lastFetchedRecord != null && request?.page != 1) {
+      querySnapshot = querySnapshot.startAfterDocument(lastFetchedRecord!);
+    }
+    querySnapshot = querySnapshot.limit(request!.pageSize);
+
+    // Get the result of the query
+    final snap = await querySnapshot.get();
+    if (request.pageSize == 10) {
+      lastFetchedRecord = snap.docs.last;
+    }
+
+    final reminders = snap.docs.asReminderList();
+
+    if (request.page == 1) {
+      final length = await remindersReference.get();
```

Comment:
Yep, this is really expensive operation

Reply:
I have replaced the fetching of the elements with fetching a counter value stored in firebase